build_scripts/build_runner.py

```diff
@@ -515,34 +568,50 @@ def extract(self):
         product_state.save_repo_states(self.default_options["PACK_DIR"] / 'repo_states.json', trigger=triggered_repo)
         shutil.copyfile(self.build_config_path, self.default_options["PACK_DIR"] / self.build_config_path.name)
 
-        for action in self.actions[Stage.EXTRACT]:
-            action.run()
+        if not self._run_build_config_actions(Stage.EXTRACT):
+            return False
 
-    def build(self):
+        return True
+
+    def _build(self):
         """
         Execute 'build' stage
 
         :return: None | Exception
         """
 
+        set_log_file(self.default_options["LOGS_DIR"] / 'build.log')
+
+        print('-' * 50)
+        self.log.info("BUILDING")
+
         self.default_options['BUILD_DIR'].mkdir(parents=True, exist_ok=True)
 
-        for action in self.actions[Stage.BUILD]:
-            action.run()
+        if not self._run_build_config_actions(Stage.BUILD):
+            return False
 
-    def install(self):
+        return True
+
+    def _install(self):
         """
         Execute 'install' stage
 
         :return: None | Exception
         """
 
+        set_log_file(self.default_options["LOGS_DIR"] / 'install.log')
+
+        print('-' * 50)
+        self.log.info("INSTALLING")
+
         self.default_options['INSTALL_DIR'].mkdir(parents=True, exist_ok=True)
 
-        for action in self.actions[Stage.INSTALL]:
-            action.run()
+        if not self._run_build_config_actions(Stage.EXTRACT):
+            return False
+
+        return True
 
-    def pack(self):
+    def _pack(self):
         """
         Pack build results
         creates *.tar.gz archives
```

Comment:
just zip and tar
what about creating zip on Linux too?

Reply:
I'll do it bit later.